### PR TITLE
Add transform pickling

### PR DESCRIPTION
**Pull request by blowekamp**
addresses #460

### CODE REVIEW

File: Wrapping/Python/Python.i [hunk 1]
```diff
@@ -653,6 +653,55 @@
 
 }
 
+
+%extend itk::simple::Transform {
+   %pythoncode %{
+
+        def __copy__(self):
+          """Create a SimpleITK shallow copy, where the internal image is shared with a copy on write implementation."""
```
Comment (zivy) [inline Wrapping/Python/Python.i]: I believe this should say "internal transform"

File: Wrapping/Python/Python.i [hunk 2]
```diff
@@ -653,6 +653,55 @@
...
+          return self.__class__(self)
+
+        def __deepcopy__(self, memo):
+          """Create a new copy of the data and internal itk::Image object."""
```
Comment (zivy) [inline Wrapping/Python/Python.i]: itk::Transform

File: Wrapping/Python/Python.i [hunk 3]
```diff
@@ -653,6 +653,55 @@
...
+          dc = self.__class__(self)
+          dc.MakeUnique()
+          return dc
+
+        def __setstate__(self, args):
+          if args[0] != 0:
+            raise ValueError("Unable to handle SimpleITK.Image pickle version {0}".args[0])
```
Comment (zivy) [inline Wrapping/Python/Python.i]: ...SimpleITK.Transform pickle version...

**Review by zivy — APPROVED** [review]
Just some minor issues where things are referred to as Image and not Transform as expected.

File: Wrapping/Python/Python.i (hunk 1 above)
Comment (blowekamp) [reply Wrapping/Python/Python.i]: done

File: Wrapping/Python/Python.i (hunk 3 above)
Comment (blowekamp) [reply Wrapping/Python/Python.i]: done

File: Wrapping/Python/Python.i (hunk 2 above)
Comment (blowekamp) [reply Wrapping/Python/Python.i]: done

**Comment by blowekamp** [comment]
Sharp is failing with linking since the ITKv5 upgrade... Need to figure out how to reproduce the problem. The Windows CI error is not related.

File: Testing/Unit/Python/ConcurrentImageRead.py [hunk 4]
```diff
@@ -28,10 +30,16 @@
...
-
     def setUp(self):
-        pass
+
+        # Create a temporary directory for output files
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+
+        # Delete the create temporary directory and files with  in. Perhaps if tests fail then the output should  stick
```
Comment (zivy) [inline Testing/Unit/Python/ConcurrentImageRead.py]: Possibly change "Delete the create temporary directory and files with   in." to "Delete the temporary files and directory."

File: Testing/Unit/Python/ImageReadWriteInt64.py [hunk 5]
```diff
@@ -27,7 +29,27 @@
...
 
-    temp_directory = ""
+    def setUp(self):
+
+        # Create a temporary directory for output files
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+
+        # Delete the create temporary directory and files with  in. Perhaps if tests fail then the output should  stick
```
Comment (zivy) [inline Testing/Unit/Python/ImageReadWriteInt64.py]: Same comment as above (it's the same text).

File: Testing/Unit/Python/sitkTransformTests.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+        for transform_type in transform_type_list:
+            self.addTypeEqualityFunc(transform_type, self.assertTransformEqual)
+
+        # Create a temporary directory for output files
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+
+        # Delete the create temporary directory and files with  in. Perhaps if tests fail then the output should  stick
```
Comment (zivy) [inline Testing/Unit/Python/sitkTransformTests.py]: Same comment as above.

**Review by zivy — APPROVED** [review]
There was one comment with respect to text, but it is in three places (same text).

**Comment by blowekamp** [comment]
@zivy The improved interface for constructing a BSplineTransform from a tuple of images, is that something that needs to be updated in the Notebooks?

**Comment by zivy** [comment]
Thanks for pointing this out, I've opened an issue for this: https://github.com/InsightSoftwareConsortium/SimpleITK-Notebooks/issues/246

**Pull request merged**